Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import CoreLocation
+import SwiftUI
+
+extension ShowDeviceLocationUsingIndoorPositioningView {
+    @MainActor
+    class Model: ObservableObject {
+        /// An IPS-aware web map for all three floors of Esri Building L in Redlands.
+        let map = Map(url: .indoorsMap)!
+        
+        /// The value of the current floor with -1 being used to represent floor that has not been set.
+        private var currentFloor: Int?
+        
+        /// The number of BLE sensors which are being used for indoor location.
+        private var sensorCount: Int?
+        
+        /// Counts the number of satellites which being used for the GPS location.
+        private var satelliteCount: Int?
+        
+        /// The value of the horizontal accuracy of the location (in meters).
+        private var horizontalAccuracy: Double?
+        
+        /// An indoors location data source based on sensor data, including but not
+        /// limited to radio, GPS, motion sensors.
+        private var indoorsLocationDataSource: IndoorsLocationDataSource?
+        
+        /// The map's location display.
+        let locationDisplay = LocationDisplay(dataSource: SystemLocationDataSource())
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextLeading: String = ""
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextTrailing: String = ""
+        
+        /// Represents loading state of indoors data, blocks interaction until loaded.
+        @Published private(set) var isLoading = false
+        
+        /// Kicks off the logic loading the data for the indoors map and indoors location.
+        func loadIndoorData() async throws {
+            isLoading = true
+            defer { isLoading = false }
+            try await map.load()
+            try await setIndoorDatasource()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task {
+                await locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// Sets the indoor datasource on the location display depending on
+        /// whether the map contains an IndoorDefinition.
```

Comment:
Not a huge deal, but the map may contain an indoor definition (instance), not an IndoorDefinition (class).

```suggestion
        /// whether the map contains an indoor definition.
```

Same elsewhere.

Note: When referring to a type or member in a documentation comment, use code voice (enclose in parentheses).